### PR TITLE
fs: Add file functions 

### PR DESCRIPTION
**Pull request by junmin-kim**
These function returns 'struct file'

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3870985f80f3b4008cb68a27e8e3e9aff96808ae] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3870985f80f3b4008cb68a27e8e3e9aff96808ae] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2967ede8205dbf343a5777755706043c0819211d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [2967ede8205dbf343a5777755706043c0819211d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a247a355e0309575e354a980a8aad8aedf4c1cf5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a247a355e0309575e354a980a8aad8aedf4c1cf5] - Code Rule Check OK.

File: os/fs/inode/fs_filedetach.c [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	do {
+		/* Take the semaphore (perhaps waiting) */
+
+		ret = sem_wait(&list->fl_sem);
+
+		/* The only case that an error should occur here is if the wait was
+		 * awakened by a signal.
+		 */
+
+		DEBUGASSERT(ret == OK || ret == -EINTR);
```
Comment (zhouxinhe) [inline os/fs/inode/fs_filedetach.c]: `ret` is OK/ERROR, but not errno.
```suggestion
		DEBUGASSERT(ret == OK || errno == EINTR);
```

File: os/fs/inode/fs_filedetach.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	} while (ret == -EINTR);
```
Comment (zhouxinhe) [inline os/fs/inode/fs_filedetach.c]:
```suggestion
	} while (errno == EINTR);
```

Please refer `_files_semtake` in fs_files.c

Comment (junmin-kim) [reply os/fs/inode/fs_filedetach.c]: I applied `_files_semtake` in `fs_files.c`

**Comment by seinfra** [comment]
Target : [4e1d1db8500c83168162d8eef0e5bae79ddc4e83] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4e1d1db8500c83168162d8eef0e5bae79ddc4e83] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [8c6756182441f64fdedd72943cb4b7382652fdba] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8c6756182441f64fdedd72943cb4b7382652fdba] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6fe72bdda3896173e9b30ed6cb4297d26386d45b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6fe72bdda3896173e9b30ed6cb4297d26386d45b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [05fc02bde1bc88164367f67efbf9418f42269f73] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [05fc02bde1bc88164367f67efbf9418f42269f73] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9f61060ad21a9d7b5796743694f2d39d60805822] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9f61060ad21a9d7b5796743694f2d39d60805822] - Code Rule Check OK.

**Pull request merged**